File: mobile/src/types/attendance.ts
import { AttendanceMethod } from './common';

export interface GenerateQRRequest {
  session_id: string;
}

export interface ValidateQRRequest {
  qr_token: string;
  session_id: string;
}

export interface AttendanceResponse {
  id: string;
  session_id: string;
  student_id: string;
  student_name: string | null;
  method: AttendanceMethod;
  recorded_at: string;
}

export interface QRTokenResponse {
  qr_image_base64: string;
  valid_until: string;
}

File: backend/app/schemas/attendance.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class GenerateQRRequest(BaseModel):
    session_id: uuid.UUID


class ValidateQRRequest(BaseModel):
    qr_token: str
    session_id: uuid.UUID


class AttendanceResponse(BaseModel):
    id: uuid.UUID
    session_id: uuid.UUID
    student_id: uuid.UUID
    student_name: str | None = None
    method: str
    recorded_at: datetime

    model_config = {"from_attributes": True}


class QRTokenResponse(BaseModel):
    qr_image_base64: str
    valid_until: datetime
